### PR TITLE
Default Kubernetes content expansion to include namespace.

### PR DESCRIPTION
**Pull request by iAnomaly**

### CODE REVIEW

**Comment by romkatv** [comment]
I don't use kubernetes myself so I'll need your help to understand why this PR is an improvement.

- Why is it better to display namespace by default together with `$P9K_KUBECONTEXT_CLOUD_CLUSTER`?
- Why is it better to display namespace by default together with `$P9K_KUBECONTEXT_NAME`?
- Why is it better to display `/default` when `kubectl` reports no namespace for the current context?

**Comment by iAnomaly** [comment]
Hi @romkatv, valid questions for sure.

- In Kubernetes, namespace is a separate isolation concept _within_ a cluster. So for the operator it is important to know not just which `$P9K_KUBECONTEXT_{CLOUD_CLUSTER|NAME}` I'm currently on but also if I've set a current namespace for that cluster.
- Because the behavior of `kubectl` is to always default to the `default` namespace unless a namespace is configured for the current context OR given explicitly with the -n, --namespace='' options, it's also helpful to know even when we are on the `default` namespace

In other words, if I use `kubectl` without any explicit `-n, --namespace=''` option, we can always know for sure what namespace our command will affect straight from Powerlevel10k.

**Comment by romkatv** [comment]
Can you imagine a situation where someone will not want to see the namespace? That is, can there be a setup in which the namespace is meaningless, perhaps because it's always the same, or because it's trivially derived from the remaining prompt content?

If I understand correctly, you are saying that `/default` must be displayed explicitly. Could you explain why it's better than not displaying `/default`?

The reason I'm asking these questions is because `kubecontext` is enabled in p10k by default, and making its format longer than necessary will annoy users. Out of 140 configs that I can find on GitHub, only one contains changes similar to what you propose. In addition, I haven't received any issues or feature requests asking for this. I hope you can see that it's not obvious in my position why the default should be changed in this specific way. I understand that for your setup this can be desired -- and you can achieve it by editing your `~/.p10k.zsh` -- but I'll need some more convincing to see that it's also a better default.

**Comment by iAnomaly** [comment]
Yes, of course I can imagine where someone will not want to see the namespace. However, my argument is that such a case is _less_ common than the case where you _would_ want to see the namespace. I don't have data on this other than my personal experience operating multiple Kubernetes clusters and supporting a team of developers who only operate a single local cluster but still want the clarity and confirmation at a glance that they are indeed on the `/default` namespace.

I'm saying that in this case, the clarity and safety of displaying the current cluster and namespace outweighs the added length in the `kubecontext` section of P10k; even for the default case of `/default` because there are users who have not internalized the `kubectl` behavior that no displayed or set namespace falls back to `/default`. In other words, you have experience with `kubectl` to know its behavior when you are not explicit.

The current P10k default `POWERLEVEL9K_KUBECONTEXT_DEFAULT_CONTENT_EXPANSION='${P9K_KUBECONTEXT_CLOUD_CLUSTER:-${P9K_KUBECONTEXT_NAME}}` was already a non-default design choice that was made. Consider if P10k behaved how `kubecontext` intended it would simply be `POWERLEVEL9K_KUBECONTEXT_DEFAULT_CONTENT_EXPANSION='${P9K_CONTENT}`. Perhaps the other 140 configs are doing this: allowing the default output of `kubecontext` to pass through unchanged?

I think "making its format longer than necessary will annoy users" is a [fallacy](https://en.wikipedia.org/wiki/Appeal_to_probability) unless users actual complain no? We could certainly revert the change if you find users are annoyed.

Thanks @romkatv. I appreciate that you take this wonderful project as seriously as you do.

Cheers,
Cameron

**Comment by romkatv** [comment]
> Perhaps the other 140 configs are doing this: allowing the default output of kubecontext to pass through unchanged?

They all use `${P9K_KUBECONTEXT_CLOUD_CLUSTER:-${P9K_KUBECONTEXT_NAME}}`. This is the default, so it's expected that almost every config should have this value. I don't know how many users out of these 140 even use kubernetes.

> We could certainly revert the change if you find users are annoyed.

I had 3 users complain about the verbose format of `name/context`. I cannot blame them -- kubernetes clusters on Google and Amazon have some incredibly long names. This is how the current code came about. It satisfied the 3 users in question, and after some time I enabled `kubecontext` by default in `.p10k.zsh`.

By the way, please don't get discouraged by my pushing back here. I take issues seriously but I simply don't consider this one. The config is flexible and you can achieve what you need by editing it. It's not a hack or workaround -- the config is meant to be edited. I do want the defaults to be good but I also prefer to err on the side of presenting less information by default, hence lots of segments such as battery, load, RAM usage must be enabled explicitly by users, and many segments that are enabled by default are tuned to be concise (various python segments, for example). If something would be considered noise by 10% of users who get exposed to it, I don't want it shown by default.

I should also note that the way you sent the PR was most unusual and immediately put me on guard. Normally people send PRs for things that will affect them, but this one will affect new users only. The PR had no description why this change is desired. 4 thumbs up appeared out of the blue immediately after the PR was sent.

> I'm saying that in this case, the clarity and safety of displaying the current cluster and namespace outweighs the added length in the `kubecontext` section of P10k; even for the default case of `/default` because there are users who have not internalized the `kubectl` behavior that no displayed or set namespace falls back to `/default`. In other words, you have experience with `kubectl` to know its behavior when you are not explicit.

I definitely don't want p10k to be more verbose by default than `kubectl`. It's not the right place to turn implicit information into explicit and to educate users about the tools they employ.

I'm thinking that maybe displaying namespace by default would be OK unless it's literally "default".  The documentation seems to imply that users who don't need namespaces should just put everything in "default". If this is the case, showing "/default" seems like a no go as it'll be 100% noise for people who don't use namespace. Namespace that isn't "default" probably has some meaning and not 100% noise.

Have you used kubernetes on gke and/or eks? If yes, how are namespaces used there? The same as on private clusters or somehow differently?

**Comment by romkatv** [comment]
> I'm thinking that maybe displaying namespace by default would be OK unless it's literally "default".

After reading some more I went ahead with this plan. If the current context's namespace is "default", it's not shown. Otherwise it is shown.
